jaxrs-tck/src/main/java/jakarta/ws/rs/tck/contextprovider/jsonb/JsonbContextProviderIT.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2021 Markus Karg. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+
+package jakarta.ws.rs.tck.contextprovider.jsonb;
+
+import static jakarta.ws.rs.RuntimeType.CLIENT;
+import static jakarta.ws.rs.core.MediaType.APPLICATION_JSON_TYPE;
+import static java.util.concurrent.TimeUnit.HOURS;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Random;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit5.ArquillianExtension;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.Timeout;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import jakarta.json.bind.Jsonb;
+import jakarta.ws.rs.client.Client;
+import jakarta.ws.rs.client.ClientBuilder;
+import jakarta.ws.rs.client.Entity;
+import jakarta.ws.rs.core.UriBuilder;
+import jakarta.ws.rs.tck.contextprovider.jsonb.EchoApplication.EchoResource;
+import jakarta.ws.rs.tck.lib.util.TestUtil;
+
+/**
+ * Compliance Test for Jsonb Context Provider of Jakarta REST API
+ *
+ * @author Markus KARG (markus@headcrashing.eu)
+ * @since 3.1
+ */
+@Timeout(value = 1, unit = HOURS)
+@ExtendWith(ArquillianExtension.class)
+public final class JsonbContextProviderIT {
+
+    @ArquillianResource
+    private URL baseUrl;
+
+    @Deployment(testable = false)
+    public static WebArchive createDeployment() {
+        WebArchive archive = ShrinkWrap.create(WebArchive.class, "jaxrs_contextprovider_jsonb_web.war");
+        archive.addClasses(EchoApplication.class, EchoResource.class, POJO.class, CustomJsonbProvider.class);
+        archive.addAsWebInfResource(JsonbContextProviderIT.class.getPackage(), "web.xml.template", "web.xml");
+
+        return archive;
+    }
+
+    @BeforeEach
+    void logStartTest(TestInfo testInfo) {
```

Comment:
Is it a convention that we *must* have these?

Reply:
I don't exactly know, but when we migrated the TCK, @alwin-joseph wrote in the migration guide to copy-paste these into every test file.

Reply:
It is not a must have. I put it in the guide when we started migration so we would have some logging of the start and end of the tests run. Please feel free to modify as you deem necessary but better to have same convention for all the tests.